vdr/doc/util.go

```diff
@@ -43,31 +43,46 @@ func IsServiceReference(endpoint string) bool {
 	return strings.HasPrefix(endpoint, "did:")
 }
 
+// DIDServiceQueryError denies the query based on validation constraints.
+type DIDServiceQueryError struct {
+	Err error // cause
+}
+
+// Error implements the error interface.
+func (e DIDServiceQueryError) Error() string {
+	return "DID service query invalid: " + e.Err.Error()
```

Comment:
isn't there a safer way which doesn't cause a panic when the inner error is nil?

Reply:
In Go pointer may not be nil, unless specified explicitly. For these rare cases it is recommended to name the field accordingly, such as `ErrOrNil`. In here, the embedded error is significant. Without it the structure/instance makes no sense.

Reply:
But nothing is preventing someone (uninitiated dev) to create such an error without the underlying error: `DIDServiceQueryError{}.Error()`

Reply:
The developer is responsible for setting all fields when creating a struct, unless the names or documentation state otherwise. Creating nil checks everywhere down to road is no solution. For example, an io.LimitReader panics when .R is nil, and that is no mistake.

If we think that our developers can not bare this responsibility, then we should use constructors. Linters notice direct struct construction when a matching New function exists.

Reply:
Agreed, and it would (should) be found by a unit test/review if it happens